### PR TITLE
ARA (Ansible Run Analysis) Role

### PR DESCRIPTION
**Pull request by rattboi**
This adds ARA to the bastion, which gives us a more in-depth view of
the our Ansible runs.

It is currently very WIP.

Signed-off-by: Bradon Kanyid <bradon@kanyid.org>

### CODE REVIEW

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/202/1489021176.32/bonnyci-run-check-multinode : FAILURE in 54s

File: roles/ara/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    state: installed
+  with_items:
+    - gcc 
+    - libffi-dev 
+    - python-dev
+    - libssl-dev
+
+- name: Install ara
+  pip:
+    name: ara
```
Comment (jamielennox) [inline roles/ara/tasks/main.yml:22]: python-app role should be able to handle this. You don't want to pip install without a virtualenv. It gets messy.

File: roles/ara/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+- name: Ensure ara directory
+  file:
+    path: /var/www/ara/
+    state: directory
+    mode: 0755
+    owner: www-data
+    group: www-data
+
+- name: Copy ARA wsgi script into place
+  command: cp -p /usr/local/bin/ara-wsgi /var/www/ara/
```
Comment (jamielennox) [inline roles/ara/tasks/main.yml]: This is odd, at very least you should symlink this instead but i don't see why it has to live in this folder.

Also for reference if you use the copy: module you can use remote_src: True to copy on the server side. Won't show changed that way.

**Review by jamielennox — DISMISSED** [review]

File: roles/ara/tasks/main.yml (hunk 1 above)
Comment (rattboi) [reply roles/ara/tasks/main.yml:22]: I initially had this going into the virtualenv that ansible-runner uses (/opt/ansible), but when I moved to the apache wsgi config, it started blowing up, so I moved it to be installed globally.

File: roles/ara/tasks/main.yml (hunk 2 above)
Comment (rattboi) [reply roles/ara/tasks/main.yml]: Yeah, I took a few shortcuts that I know I need to go back to and clean up. I'll do that shortly.

File: roles/ara/tasks/main.yml (hunk 1 above)
Comment (rattboi) [reply roles/ara/tasks/main.yml:22]: python-app doesn't work here, because python-app only does git-based installs, no?

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/202/1489082267.16/bonnyci-run-check-multinode : FAILURE in 9s

File: roles/ara/tasks/main.yml (hunk 1 above)
Comment (jamielennox) [reply roles/ara/tasks/main.yml:22]: we can make that optional, just have python-app install a series of things into a virtualenv in a known location.

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/202/1489110178.37/bonnyci-run-check-multinode : RETRY_LIMIT

**Comment by jamielennox** [comment]
recheck

File: roles/ara/tasks/main.yml (hunk 1 above)
Comment (rattboi) [reply roles/ara/tasks/main.yml:22]: I'm fine with that, but I've now got it installing into the venv again using the pip module.

**Review by omgjlk — COMMENTED** [review]
I don't see anything wrong here, but will let Jamie give the final word.

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/202/1489192182.32/bonnyci-run-check-multinode : FAILURE in 6m 56s

**Comment by rattboi** [comment]
recheck

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/202/1489192672.38/bonnyci-run-check-multinode : FAILURE in 6m 55s

**Comment by rattboi** [comment]
Ok, ready for real review. Works well in my testing.

**Comment by rattboi** [comment]
recheck

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/202/1489193907.3/bonnyci-run-check-multinode : FAILURE in 6m 57s

**Comment by gandelman-a** [comment]
recheck

**Comment by gandelman-a** [comment]
recheck

**Comment by gandelman-a** [comment]
recheck

**Comment by gandelman-a** [comment]
recheck

**Comment by gandelman-a** [comment]
recheck

**Comment by gandelman-a** [comment]
recheck

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/202/1489210099.03/bonnyci-run-check-multinode : FAILURE in 7m 02s

**Comment by jamielennox** [comment]
recheck

**Review by omgjlk — COMMENTED** [review]
LGTM, but I'll let somebody else approve it.

**Comment by gandelman-a** [comment]
This looks good from an ansible POV, but I'll let someone who can test locally flip the final approval.

We may want to consider storing ARA data in a mysql database or some other remote database.  We have the per-bonny env mysql servers running on the zuul hosts but I'm not sure that'd be the best place.

**Review by ConsoleCatzirl — APPROVED** [review]

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/gate_github/BonnyCI/hoist/202/1489435098.45/bonnyci-run-gate-multinode : FAILURE in 33s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/gate_github/BonnyCI/hoist/202/1489436895.79/bonnyci-run-gate-multinode : FAILURE in 1m 37s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/gate_github/BonnyCI/hoist/202/1489440061.13/bonnyci-run-gate-multinode : FAILURE in 1m 22s

**Pull request merged**